receiver/awsxrayreceiver/internal/translator/translator.go

```diff
@@ -0,0 +1,190 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package translator
+
+import (
+	"encoding/json"
+
+	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/awsxrayreceiver/internal/tracesegment"
+)
+
+const (
+	// just a guess to avoid too many memory re-allocation
```

Comment:
Unless this guess is x-ray specific I'd leave it out and use default, otherwise document what the guess is based on

Reply:
It's not x-ray specific. However, there's no default for https://pkg.go.dev/go.opentelemetry.io/collector@v0.7.0/consumer/pdata?tab=doc#AttributeMap. Either we just initialize with size 0 (and potentially experience reallocation a few times when we insert number of attributes) or a non-zero guess to avoid re-allocation before we have 15 attributes.